Write end-user manual (#410)
Uninett/Argus-frontend: MANUAL.md

# User's guide to Argus
- [About Argus](#what-is-argus)
- [Log in](#log-into-argus)
  - [Using username and password](#login-using-username-and-password)
  - [Using federated login](#login-using-federated-login-feide-in-the-example-below)
- [Manage alarms](#work-with-alarms-in-argus)

>   - [What is incident](#what-is-an-incident-in-argus)

Reviewer: ````suggestion
  - [What is an incident](#what-is-an-incident-in-argus)
````

    - [Close (resolve) incident](#close-resolve-an-incident)
    - [Acknowledge incident](#add-acknowledgement-to-an-incident)
    - [Add ticket to incident](#add-ticket-url-to-an-incident)
    - [Remove ticket from incident](#remove-ticket-url-from-an-incident)
  - [Update several incidents](#update-several-incidents-at-a-time)
    - [Re-open incidents](#re-open-closed-resolved-incidents)
    - [Close incidents](#close-resolve-incidents)
    - [Acknowledge incidents](#add-acknowledgement-to-incidents)
    - [Add ticket to incidents](#add-ticket-url-to-incidents)
    - [Remove ticket from incidents](#remove-ticket-url-from-incidents)
- [Customize notifications](#customize-alarm-notifications-in-argus)
  - [Access your notification settings](#access-your-notification-profiles)
  - [Add notification profile](#add-new-notification-profile)
  - [Edit notification profile](#edit-existing-notification-profile)
  - [Disable notification profile](#disable-notification-profile)
  - [Delete notification profile](#delete-notification-profile)
- [Manage notification time](#manage-when-to-receive-notifications-in-argus)

>   - [What is timeslot](#what-is-a-timeslot-in-argus)
>   - [What is recurrence](#what-is-a-recurrence-in-argus)

Reviewer: ````suggestion
  - [What is a timeslot](#what-is-a-timeslot-in-argus)
  - [What is a recurrence](#what-is-a-recurrence-in-argus)
````

  - [Edit timeslot](#edit-existing-timeslot)
  - [Delete timeslot](#delete-timeslot)
- [Manage contact details](#manage-your-contact-details-in-argus)
  - [Access your contact details](#access-your-contact-details-in-settings)
  - [Add phone number](#add-new-phone-number-in-settings)
  - [Edit phone number](#edit-existing-phone-number-in-settings)
  - [Delete phone number](#delete-phone-number-in-settings)
- [Log out](#log-out-from-argus)

## What is Argus?
Argus is an _alert aggregator_ designed for storing and managing alerts from different monitoring systems at one place.
Argus is created for **ease of alarm management** and **customizable alarm notifications**.

## Log into Argus
Argus supports several login mechanisms:
* _username-password login_
* _federated login with OAuth 2.0_

Log in and start using Argus at **/login**.
<img src="public/screenshots/manual/Screenshot 2022-09-08 at 13.49.23.png"/>

### Login using username and password
1. Fill out _username_ and _password_.

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 13.48.07.png"/>

2. Press `LOGIN`.

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 13.48.07 2.png"/>

> ### Login using federated login (Feide in example)

Reviewer: ````suggestion
### Login using federated login (Feide for example)
````

> ### Login using federated login (Feide in example)

Reviewer: I like that phrasing, that clears it up

#### Change how often incidents table gets refreshed
1. Click on the _gears icon_ to the right below the header.

   <img src="public/screenshots/manual/Screenshot 2022-09-09 at 10.08.23.png"/>
   
2. Select refresh method in the _Auto Update selector_:

   * If you want **no automatic table updates**, press `NEVER` in the _Auto Update selector_. Note that you will have to refresh the page yourself if you want the table to get updated.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-09 at 10.08.45.png"/>
     
   * If you want the table to update **in realtime**, press `REALTIME` in the _Auto Update selector_.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-09 at 10.08.45 copy.png"/>
     
   * If you want the table to get updated **every couple of seconds**, press `INTERVAL` in the _Auto Update selector_.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-09 at 10.08.45 copy 2.png"/>
      
     * You can see the value of the current refresh interval below the _Incidents table_. The refresh interval is displayed **in seconds**.

       <img src="public/screenshots/manual/Screenshot 2022-09-22 at 08.16.17.png"/>
     

>      * You can change the value of a refresh interval in `/src/config.tsx`. The refresh interval is stored **in seconds**.

Reviewer: ````suggestion
     * You can change the value the refresh interval in `/src/config.tsx`. The refresh interval is stored **in seconds**.
````

Reviewer: But there is only one interval if I understand it correctly

Reviewer 2: "value of" doesn't mean the same as "value".

Reviewer 2: Considering the previous sentence the best solution is probably "You can change the refresh interval value in `/src/config.tsx`." Also, I assume the refresh value is *whole seconds*, not a float?

After you [have set the preferred filter parameters for incidents](#decide-which-incident-are-shown-in-the-table), you can save your preferences as a _filter_. Stored _filters_ can be used when [customizing alarm notifications](#customize-alarm-notifications-in-argus).

#### Save current filter
1. [Set the preferred filter parameters](#decide-which-incident-are-shown-in-the-table).
2. Click on the _plus icon_ within the _Filter input field_.

   <img src="public/screenshots/manual/Screenshot 2022-09-09 at 14.30.58.png"/>
    
3. Give a (meaningful) name to your filter. Press `CREATE`. Note that you can not edit a filter's name after it is created.

   <img src="public/screenshots/manual/Screenshot 2022-09-09 at 14.31.38.png"/>

#### Modify existing filter
1. [Make desired changes to filter parameters](#decide-which-incident-are-shown-in-the-table).
2. Click on the _save icon_ within the _Filter input field_.

   <img src="public/screenshots/manual/Screenshot 2022-09-09 at 14.33.09.png"/>

3. Click on the filter that you want to update, and press `SAVE TO`.

   <img src="public/screenshots/manual/Screenshot 2022-09-09 at 14.36.22.png"/>

#### Apply existing filter

> 1. Click into the _Filter input field_.

Reviewer: ````suggestion
1. Click on the _Filter input field_.
````

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.42.35.png"/>
    
3. Press `SAVE`.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.44.40.png"/>
    

#### Remove ticket URL from an incident
1. [Open incident in detailed view](#access-detailed-incident-view).
2. Remove URL from the _Ticket input field_ and press `SAVE`.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.44.59.png"/>
    

### Update several incidents at a time

#### Re-open closed (resolved) incidents
1. Select several incidents in the _Incidents table_ and press `RE-OPEN SELECTED` in the _table toolbar_.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.18.39.png"/>
    
2. Confirm re-opening.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.40.51.png"/>

#### Close (resolve) incidents
1. Select several incidents in the _Incidents table_ and press `CLOSE SELECTED` in the _table toolbar_.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.19.15.png"/>
    
2. Press `CLOSE NOW`. Note that you can optionally provide a closing comment if needed.

>    <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.46.05.png"/>

Reviewer: I didn't mean in the manual itself, but in the screenshot (in grey it says "Write a message describing why the incidents were manually closed" where the user will enter the closing comment)

>    <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.46.05.png"/>

Author: Oh, I see! I will create an issue on fixing helper text for bulk-operations.

#### Add acknowledgement to incidents
1. Select several incidents in the _Incidents table_ and press `ACK` in the _table toolbar_.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.19.15 copy.png"/>
    
2. Press `SUBMIT`. Note that you can optionally provide an acknowledgement comment and/or date when these acknowledgements are no longer relevant.

>    <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.46.26.png"/>

Author: Not sure what you mean here..? 😅

>    <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.46.26.png"/>

Reviewer: The same as in line 418

>    <img src="public/screenshots/manual/Screenshot 2022-09-12 at 07.46.26.png"/>

Author: I will create a separate issue on fixing helper text for bulk-operations.

#### Add ticket URL to incidents
1. Select several incidents in the _Incidents table_ and press `ADD TICKET` in the _table toolbar_.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.20.10.png"/>
    
2. Type in/paste in ticket URL into the _Valid ticket URL field_ and press `SUBMIT`. Note that URL has to be absolute (full website address).

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.20.59.png"/>
    

#### Remove ticket URL from incidents
1. Select several incidents in the _Incidents table_ and press `ADD TICKET` in the _table toolbar_.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.20.32.png"/>

2. Leave the _Valid ticket URL field_ empty and press `SUBMIT`.

   <img src="public/screenshots/manual/Screenshot 2022-09-12 at 08.20.59.png"/>

## Customize alarm notifications in Argus
Choose **when**, **where** and **what** alarm notifications you want to receive by creating, editing and deleting _notification profiles_.

### Access your notification profiles
1. Press `PROFILES` in the header.

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 13.53.49 2.png"/>

> ### Add new notification profile

Author: The whole section about notification profiles is based on the _old_ version of notification profiles, and will be updated when we roll-out v.2 in production. That's why there are no screenshots here.

> ### Add new notification profile

Reviewer: That's good!

1. Select timeslot for when to receive notifications in _Timeslot drop-down_. If drop-down menu is empty, [create a timeslot](#add-new-timeslot) first.
2. Select what alarms you want to receive notifications about in _Filter drop-down_. If drop-down menu is empty, [create a filter first](#save-current-filter).
3. Choose whether you want to receive notifications via SMS or Email in _Media drop-down_.

> 4. Select what phone number(s) you want to receive notifications to in _Phone number drop-down_. If drop-down menu is empty, [add a phone number](#add-new-phone-number-in-settings) first.

Author: The whole section about notification profiles is based on the _old_ version of notification profiles, and will be updated when we roll-out v.2 in production. Optional phone numbers is the first PR that is marked with _API v.2_ label. So I don't think that mentioning it here is relevant since in API v.1 SMS is the only notification type. Makes no sense to create notification without notification medium, or? 🤷‍♀️

> 4. Select what phone number(s) you want to receive notifications to in _Phone number drop-down_. If drop-down menu is empty, [add a phone number](#add-new-phone-number-in-settings) first.

Reviewer: But Email is also a notification medium
Now that I think about it it would be good to let the user know that if they select email the email will be sent to their email address.
Is there a way for the user to find out which email address they are registered with?

> 4. Select what phone number(s) you want to receive notifications to in _Phone number drop-down_. If drop-down menu is empty, [add a phone number](#add-new-phone-number-in-settings) first.

Reviewer: But I agree with not mentioning yet that phone numbers are optional

> 4. Select what phone number(s) you want to receive notifications to in _Phone number drop-down_. If drop-down menu is empty, [add a phone number](#add-new-phone-number-in-settings) first.

Author: Is it a notification medium in API v.1?

5. Press `CREATE`.

> ### Edit existing notification profile

Author: The whole section about notification profiles is based on the old version of notification profiles, and will be updated when we roll-out v.2 in production. That's why there are no screenshots here.

1. Change timeslot for when to receive notifications in _Timeslot drop-down_ (if needed).
2. Change what alarms you want to receive notifications about in _Filter drop-down_ (if needed).
3. Change whether you want to receive notifications via SMS or Email in _Media drop-down_ (if needed).
4. Change what phone number(s) you want to receive notifications to in _Phone number drop-down_ (if needed).
5. Press `SAVE`.

> ### Disable notification profile

Author: The whole section about notification profiles is based on the old version of notification profiles, and will be updated when we roll-out v.2 in production. That's why there are no screenshots here.

1. Uncheck the _Active checkbox_ inside one of your existing notification profiles.

> ### Delete notification profile

Author: The whole section about notification profiles is based on the old version of notification profiles, and will be updated when we roll-out v.2 in production. That's why there are no screenshots here.

1. Press `DISCARD` inside one of your existing notification profiles.

## Manage when to receive notifications in Argus
Add, edit or delete timeslots in _Timeslots_.

<img src="public/screenshots/manual/Screenshot 2022-09-08 at 14.02.43.png"/>
<img src="public/screenshots/manual/Screenshot 2022-09-08 at 14.03.02.png"/>

### What is a timeslot in Argus
A timeslot is a collection of one or more recurrences with a meaningful name. Saved timeslots can be used when [customizing alarm notifications](#customize-alarm-notifications-in-argus). Each timeslot represents a window (or several windows) of time for when it is OK to receive alarm notifications.

### What is a recurrence in Argus

> Recurrences are a building blocks for timeslots. Each recurrence represents a time range on selected weekdays for when it is OK to receive alarm notifications.

Reviewer: ````suggestion
Recurrences are building blocks for timeslots. Each recurrence represents a time range on selected weekdays for when it is OK to receive alarm notifications.
````

   * If needed, change _start time_ either by typing a new value or by using the calendar icon.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.03.51.png"/>
     
   * If needed, change _end time_ either by typing a new value or by using the calendar icon.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.05.42.png"/>
     
   * Check _All day_ if you want the recurrence to be from 00:00 a.m. to 11:59 p.m. Note that if _All day_ is checked, you do not need to provide _start-_ and _end time_.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.07.27.png"/>
     
   * If needed, change day(s):
     1. Open drop-down menu.
     
        <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.05.42 2.png"/>
        
     2. Select/de-select days for this recurrence by clicking on them once. Selected days are highlighted in light-yellow.
     
        <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.08.24.png"/>
        
     3. Click away anywhere outside the drop-down menu.

### Delete recurrence
1. [Go to your timeslots](#access-your-timeslots).

> 2. Press `REMOVE` inside one of the existing recurrences either in _Create New Timeslot_ box, or inside one of your existing timeslots.

Reviewer: ````suggestion
2. Press `REMOVE` inside one of the existing recurrences either in the _Create New Timeslot_ box, or inside one of your existing timeslots.
````

   
8. _Create New Timeslot_ box will refresh to default and your newly created timeslot will appear at the bottom of the timeslot list. Note that existing timeslots have a dark border at the top.

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.41.07.png"/>

### Edit existing timeslot
1. [Go to your timeslots](#access-your-timeslots).
2. Modify one of your existing timeslots:
   * Change name if needed.
   
     <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.47.38.png"/>
     
   * [Add another recurrence(s)](#add-new-recurrence) if needed.
   * [Edit recurrence(s)](#edit-recurrence) if needed.
   * [Remove recurrence(s)](#delete-recurrence) if needed.
3. Press `SAVE`. Note that `SAVE`-button is inactive if no changes were made. `SAVE`-button is also inactive if some changes are invalid. In this case error messages inside the timeslot box will help you.

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.49.29.png"/>

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.53.44.png"/>

### Delete timeslot
1. [Go to your timeslots](#access-your-timeslots).

> 2. Press `DELETE` inside one of the existing timeslots. Note that `DELETE`-button is disabled in the _Create New Timeslot_ box.

Reviewer: ````suggestion
2. Press `DELETE` inside one of the existing timeslots. Note that the `DELETE`-button is disabled in the _Create New Timeslot_ box.
````

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 12.56.02.png"/>

## Manage your contact details in Argus
Add, edit or delete phone numbers in your settings. Phone numbers that are present in your settings can be used when [customizing alarm notifications](#customize-alarm-notifications-in-argus).

<img src="public/screenshots/manual/Screenshot 2022-09-08 at 14.00.29.png"/>

### Access your contact details in settings

> 1. Click on _user icon_ in the header.

Reviewer: ````suggestion
1. Click on the _user icon_ in the header.
````

   <img src="public/screenshots/manual/Screenshot 2022-09-08 at 13.53.49.png"/>
   
2. Click on `Settings` in the drop-down menu.

    <img src="public/screenshots/manual/Screenshot 2022-09-07 at 14.18.11.png"/>
    
### Add new phone number in settings
1. [Go to your contact details](#access-your-contact-details-in-settings).

> 2. Type in an existing phone number. Phone number **has to include land code**.

Reviewer: ````suggestion
2. Type in an existing phone number. The phone number **has to include the country code**.
````
